games/devtest/mods/unittests/async_env.lua

```diff
@@ -166,3 +166,43 @@ local function test_userdata_passing2(cb, _, pos)
 	end, vm, pos)
 end
 unittests.register("test_userdata_passing2", test_userdata_passing2, {map=true, async=true})
+
+local function test_async_metatable_registration(cb)
+	if not pcall(core.register_async_metatable, "__builtin:vector", vector.metatable) then
+		error("Metatable name aliasing throws an error when it should be allowed")
+	end
+
+	if pcall(core.register_async_metatable, "__builtin:vector", {}) then
+		error("Illegal metatable overriding allowed")
```

Comment:
actually: you can't use `assert` or `error` in async tests

this will have to be adjusted.

Reply:
Seems like the `async_metatable_registration` test does not involve the async environment. I left the `cb` argument there for some reason.